swift-sdk/EmbeddedHelper.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  Copyright © 2023 Iterable. All rights reserved.
+//
+import UIKit
+
+/// Utility Methods for embedded
+/// All classes/structs are internal.
+struct EmbeddedHelper {
+    enum EmbeddedClickedUrl {
+        case localResource(name: String) // applewebdata://abc-def/something => something
+        case iterableCustomAction(name: String) // iterable://something => something
+        case customAction(name: String) // action:something => something or itbl://something => something
+        case regularUrl(URL) // protocol://something => protocol://something
+    }
+
+    static func parse(embeddedUrl url: URL) -> EmbeddedClickedUrl? {
+        guard let scheme = UrlScheme.from(url: url) else {
+            ITBError("Request url contains an invalid scheme: \(url)")
+            return nil
+        }
+
+        switch scheme {
+        case .applewebdata:
+            ITBError("Request url contains an invalid scheme: \(url)")
+            guard let urlPath = getUrlPath(url: url) else {
+                return nil
+            }
+            return .localResource(name: urlPath)
+        case .iterable:
+            return .iterableCustomAction(name: dropScheme(urlString: url.absoluteString, scheme: scheme.rawValue))
+        case .action, .itbl:
+            return .customAction(name: dropScheme(urlString: url.absoluteString, scheme: scheme.rawValue))
```

Comment:
@evantk91 Is it possible here that the the action depends on the specific scheme that prefixes it? Maybe there are multiple different schemes supported by the app? Not sure it's safe to drop it?